src/main/java/school/faang/user_service/S3/S3ServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package school.faang.user_service.S3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import school.faang.user_service.entity.User;
+import school.faang.user_service.entity.UserProfilePic;
+import org.springframework.beans.factory.annotation.Value;
+import school.faang.user_service.repository.UserRepository;
+
+import javax.imageio.ImageIO;
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+@Service
+public class S3ServiceImpl implements S3Service {
+
+    private static final Logger logger = LoggerFactory.getLogger(S3ServiceImpl.class);
+
+    private final AmazonS3 amazonS3Client;
+    private final String bucketName;
+    private final UserRepository userRepository;
+
+    @Autowired
+    public S3ServiceImpl(AmazonS3 amazonS3Client, @Value("${services.s3.bucketName}") String bucketName, UserRepository userRepository) {
+        this.amazonS3Client = amazonS3Client;
+        this.bucketName = bucketName;
+        this.userRepository = userRepository;
+    }
+
+    // Метод загрузки файла в Amazon S3 и связывания его с пользователем
+    @Override
+    public void uploadFileForUser(MultipartFile file, String folder, Long userId) throws IOException {
```

Comment:
I guess it is redundant here to have "throws IOException" on your method, as you are using try-catch block